### PR TITLE
🏗  🚮  Remove `test-status` bot reporting

### PR DESCRIPTION
**Pull request by rileyajones**
Since moving to CircleCI, the `test-status` bot has become more of a maintenance burden than a help due to the CircleCI GitHub integration providing the same information.

#35247

### CODE REVIEW

File: src/global.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+declare global {
```
Comment (danielrozenberg) [inline src/global.d.ts:1]: Hmm?

File: testing123 [hunk 2]
```diff
@@ -0,0 +1 @@
+hello world
```
Comment (danielrozenberg) [inline testing123:1]: Debugging leftover?

**Review by danielrozenberg — APPROVED** [review]
LGTM after removing the debugging leftovers. Goodnight, sweet bot!

File: src/global.d.ts (hunk 1 above)
Comment (rileyajones) [reply src/global.d.ts:1]: Debug leftover :/

File: build-system/tasks/report-test-status.js [hunk 3]
```diff
@@ -1,236 +0,0 @@
...
-const argv = require('minimist')(process.argv.slice(2));
-const fetch = require('node-fetch');
-const {
-  isCircleciBuild,
-  isGithubActionsBuild,
-  isPullRequestBuild,
-} = require('../common/ci');
-const {ciJobUrl} = require('../common/ci');
-const {cyan, yellow} = require('../common/colors');
-const {getValidExperiments} = require('../common/utils');
```
Comment (rsimha) [inline build-system/tasks/report-test-status.js:27]: Looks like this is the only spot where `getValidExperiments` was being used. If so, it too can be deleted from  `utils.js`.

**Review by rsimha — APPROVED** [review]
Thanks for the cleanup. LGTM after one more deletion.

And appreciate all your past work on this, @danielrozenberg!

File: build-system/tasks/report-test-status.js (hunk 3 above)
Comment (rileyajones) [reply build-system/tasks/report-test-status.js:27]: Excellent catch

**Review by estherkim — APPROVED** [review]
Did you want to delete https://github.com/ampproject/amp-github-apps/tree/main/test-status too? 😬 

And we can turn down the app engine

**Comment by rileyajones** [comment]
@estherkim we're planning on waiting a couple days before turning down the bot incase someone complains (can't really imagine that happening).

I'll put up a PR deleting the code in `amp-github-apps` and link it here though.

**Pull request merged**